From 41148b969fcb5e775d28361a92ffd1ef780627b6 Mon Sep 17 00:00:00 2001
From: Hariharan K <quic_harihk@quicinc.com>
Date: Sun, 18 Aug 2024 20:39:38 +0530
Subject: [PATCH] i2c: qcom-geni: Update driver sequence

Calling firmware load without turning on the
clocks causes kernel to hang. Hence Updated
the driver sequence to call the firmware load
after the SE clocks are turned ON.

Change-Id: If96805547348974874eb3dc27517385f94d5a5ab
Signed-off-by: Hariharan K <quic_harihk@quicinc.com>
(cherry picked from commit f266c4096f156be6651678a1dc739fc83cf93d70)
---
 drivers/i2c/busses/i2c-qcom-geni.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/i2c/busses/i2c-qcom-geni.c b/drivers/i2c/busses/i2c-qcom-geni.c
index 85edb2fd8bcf..e612bfaf7ec0 100644
--- a/drivers/i2c/busses/i2c-qcom-geni.c
+++ b/drivers/i2c/busses/i2c-qcom-geni.c
@@ -811,9 +811,6 @@ static int geni_i2c_probe(struct platform_device *pdev)
 		gi2c->clk_freq_out = KHZ(100);
 	}
 
-#ifdef CONFIG_QCOM_GENI_SE_FW_LOAD
-	geni_se_fw_load(&gi2c->se, QUPV3_SE_I2C);
-#endif /* CONFIG_QCOM_GENI_SE_FW_LOAD */
 	if (has_acpi_companion(dev))
 		ACPI_COMPANION_SET(&gi2c->adap.dev, ACPI_COMPANION(dev));
 
@@ -872,6 +869,11 @@ static int geni_i2c_probe(struct platform_device *pdev)
 		dev_err(dev, "Error turning on resources %d\n", ret);
 		return ret;
 	}
+
+#ifdef CONFIG_QCOM_GENI_SE_FW_LOAD
+	geni_se_fw_load(&gi2c->se, QUPV3_SE_I2C);
+#endif /* CONFIG_QCOM_GENI_SE_FW_LOAD */
+
 	proto = geni_se_read_proto(&gi2c->se);
 	if (proto != GENI_SE_I2C) {
 		dev_err(dev, "Invalid proto %d\n", proto);
-- 
2.34.1

